Bottom: 7877a1a5ec8b625a7127ef30ba236e4677df5df2
Top:    c3cf2f5759614f2ebce8277cf3fac4f7c1643d05
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-18 11:42:39 -0700

Refresh of fork

---

diff --git a/src/preload.c b/src/preload.c
index a1073d5..5b6b50a 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -468,8 +468,8 @@ static int connect_fork(int socket, const struct sockaddr *addr, socklen_t addrl
 	long flags;
 
 	fd = fd_getd(socket);
-	flags = real.fcntl(fd, F_GETFD);
-	real.fcntl(fd, F_SETFD, 0);
+	flags = real.fcntl(fd, F_GETFL);
+	real.fcntl(fd, F_SETFL, 0);
 	ret = real.connect(fd, addr, addrlen);
 	if (ret)
 		return ret;
@@ -480,7 +480,7 @@ static int connect_fork(int socket, const struct sockaddr *addr, socklen_t addrl
 		return 0;
 	}
 
-	real.fcntl(fd, F_SETFD, flags);
+	real.fcntl(fd, F_SETFL, flags);
 	ret = transpose_socket(socket, fd_rsocket);
 	if (ret < 0)
 		return ret;
